src/index.jsx

```diff
@@ -6,13 +6,14 @@ import GitUrlParse from 'git-url-parse';
 
 import { useState } from 'react';
 
-import { Button, Box, Text, Popover, Heading, ThemeProvider, TextInput } from '@primer/components';
-import { CopyIcon } from '@primer/octicons-react';
+import { Button, Box, Text, Popover, Heading, ThemeProvider, TextInput, SelectMenu, DropdownMenu, DropdownButton } from '@primer/components';
+
+import { CopyIcon, TabExternalIcon } from '@primer/octicons-react';
 
 import { AVAILABLE_APPS, generateRegularUrl } from './generator';
 import { getPref, setPref } from './prefs';
 
-function copyGeneratedUrl(hubUrl, app) {
+function copyGeneratedUrl(hubUrl, app, open) {
```

Comment:
This function now is doing too many things - figuring out the correct URL, putting it into the clipboard, and optionally opening a new tab. This should be split into three things:

1. A `generateUrl` function, that returns the generated URL
2. Something that just calls `navigator.clipboard.writeText` with the return value of (1). This can also be in the click handler of the appropriate button
3. Something that just calls `chrome.tabs.create` with the return value of (1). This can also be the click handler of the appropriate button.

This also prevents the current behavior, where opening this in a new tab clobbers your clipboard, which is not really intuitive behavior.

Reply:
fixed in new PR